### PR TITLE
Add internal set method

### PR DESCRIPTION
**Pull request by BabkinAleksandr**
Working on implementation of internal `set` method.

## About:
The `_set` is consist of two internal structures:
- **`collection`**: array, that keeps all finally filtered values, that are available through `values()` method.
- **`set`**: object, which keys is used for first compare. Every key is keeping array of `collection` indexes. All values under indexes passes first comparing. These indexes is used for ref checking for structure items

### Comparing

All items finally are compared with `===`.

First internal comparing is checking if internal `set` has such `key`. If has, array under this key is iterating through it's keys and comparing `===` every value with provided.

### Props:
`size` - get the `set` values length

### Methods:
`add`: adds new value to set
`concat`: concats set with provided list
`values`: returns the array of values inside `set`
`has`: return boolean value whether values is in `set` or not

## TO DO:
- [x] ~~add `delete` method~~ _i've made some tests, it would slow down `_set` and I doubt we would use it_
- [ ] reduce size _(tried, but all reduces are lower performance)_
- [x] add `NaN` support
- [x] optimize performance
   - [x] ~~come up with handling plain items. Maybe care it as key with type like `[value + typeof value]`~~ _I've made tests: it's more complicated (and slower) to check types every time than just loops through array every time_

----------

I've run the test for `union`, it looks much better:

aleksandrbabkin_aleksandrs-macbook-pro____documents_nanoutils__zsh_

### CODE REVIEW

**Comment by BabkinAleksandr** [comment]
Speeded up `_set` and used it in `union` method:

project_ ___documents_nanoutils

**Comment by BabkinAleksandr** [comment]
@Beraliv @Kelin2025 I think you could check it, after that I will refactor here all method, that requires `_set` and check performance

File: lib/_internal/_set/index.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  if (value + '' === 'NaN') return 0
+  for (var i = 0; i < set[value].length; i++) {
+    if (collection[set[value][i]] === value) return set[value][i]
+  }
+  return -1
+}
+
+function addValue(value, set, collection) {
+  var index = getValueIndex(value, set, collection)
+  if (index >= 0) return
```
Comment (Beraliv) [inline lib/_internal/_set/index.js]: try `~` operator, it's shorter but I'm not sure, it's quicker

File: TIMES.md [hunk 2]
```diff
@@ -145,24 +145,24 @@
...
-|                     | ramda |    0.83ms |    0.60ms |    0.47ms |
-|                     | diff  |  +26.52ms |  +15.38ms |   +8.38ms |
-|                     |       | +3214.24% | +2553.37% | +1799.79% |
-| ------------------- | ----- | --------- | --------- | --------- |
+| Method              | Lib   |       0% |      50% |      100% |
+| ------------------- | ----- | -------- | -------- | --------- |
+| intersection        | nano  |  18.49ms |   0.07ms |    0.03ms |
+|                     | ramda |   1.81ms |   0.76ms |    0.61ms |
+|                     | diff  | +16.67ms |  -0.69ms |   -0.58ms |
+|                     |       | +919.59% | -986.13% | -2089.57% |
```
Comment (Beraliv) [inline TIMES.md]: I think intersection should use another way or combine different approaches

File: lib/_internal/_set/index.js (hunk 1 above)
Comment (Beraliv) [reply lib/_internal/_set/index.js]: Ahhh, yes, you're right

**Pull request merged**